scripts/linters/pylint_extensions.py

```diff
@@ -1959,6 +2018,244 @@ def process_tokens(self, tokens):
+
+            # Checking if multi-line import is present.
+            # Eg: from typing import (
+            #   Any, Callable, Dict, FrozenSet, Iterator, List, Set,
+            #   Tuple, Type, cast
+            # )
+            if token_type == tokenize.OP:
+                if single_line_import and token == '(':
+                    multi_line_import = True
+                    single_line_import = False
+                if multi_line_import and token == ')':
+                    multi_line_import = False
+
+            if token_type == tokenize.NAME and token == 'cast':
+                # Passing those cases where cast is imported.
+                if single_line_import and import_line_num == line_num:
+                    pass
+                elif multi_line_import:
+                    pass
+                # Throwing an error when cast is encountered but there is no
+                # corresponding comment exist.
+                elif cast_comment_present:
+                    cast_comment_present = False
+                else:
+                    self.add_message(
+                        'cast-func-used', line=line_num)
+
+    def check_comment_is_present_with_any_type(self, tokens):
+        """Checks whether the Any type in a module has been documented
+        or not. If the Any type is not documented then adds an error
+        message.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+        """
+        any_type_comment_present = False
+        outside_function_def = True
+        outside_args_section = True
+        any_present_inside_arg_section = False
+        any_present_inside_return_section = False
+        any_present_in_function_def = False
+        multi_line_import = False
+        single_line_import = False
+        import_line_num = 0
+        args_section_end_line_num = 0
+
+        any_type_regex = r'^# Here we used type Any because'
+
+        for (token_type, token, (line_num, _), _, line) in tokens:
+            line = line.strip()
+
+            if token_type == tokenize.COMMENT:
+                if re.search(any_type_regex, line):
+                    any_type_comment_present = True
+
+            # Checking if single-line import is present.
+            # Eg: from typing import Any.
+            if token_type == tokenize.NAME:
+                if token == 'import':
+                    single_line_import = True
+                    import_line_num = line_num
+
+            # Checking if multi-line import is present.
+            # Eg: from typing import (
+            #   Any, Callable, Dict, FrozenSet, Iterator, List, Set,
+            #   Tuple, Type, cast
+            # )
+            if token_type == tokenize.OP:
+                if single_line_import and token == '(':
+                    multi_line_import = True
+                    single_line_import = False
+                if multi_line_import and token == ')':
+                    multi_line_import = False
```

Comment:
Slight repetition of code, can it be separated into a helper function?

Reply:
Here we are checking for every token and maintaining some variables ( multi_line_import, single_line_import ) to keep track the status of import. If we shift these lines inside a function then every time we call the function it will reinitialize all these variables and lose the previous data of import.

Reply:
Grouping these to a dict or any mutable type and passing it to a helper function wouldn't be a problem, right?

Reply:
import checks are separated into a helper function, PTAL!